react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { activeIndex: 0 };
+    this.revolver = this.revolver.bind(this);
+    this.nextImg = this.nextImg.bind(this);
+    this.prevImg = this.prevImg.bind(this);
+  }
+
+  revolver() {
+    this.intervalId = setInterval(this.nextImg, 3000);
+  }
+
+  nextImg() {
+    this.setState({ activeIndex: this.state.activeIndex + 1 });
+    if (this.state.activeIndex === this.props.images.length - 1) {
+      this.setState({ activeIndex: 0 });
+    }
+  }
```

Comment:
`setState` should be the last thing to happen in the method:
```JAVASCRIPT
nextImg() {
  let activeIndex = this.state.activeIndex + 1;
  if (activeIndex >= this.props.images.length - 1) {
    activeIndex = 0;
  }
  this.setState({ activeIndex: activeIndex });
}
```

Reply:
That makes much more sense.  Thank you Scott!